aws/resource_aws_ssm_patch_baseline.go

```diff
@@ -22,14 +25,23 @@ func resourceAwsSsmPatchBaseline() *schema.Resource {
 		},
 
 		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
 			"name": {
 				Type:     schema.TypeString,
 				Required: true,
+				ValidateFunc: validation.Any(
+					validation.StringLenBetween(3, 128),
+					validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9_\-.]{3,128}$`), ""),
+				),
 			},
 
 			"description": {
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1024),
```

Comment:
To prevent issues because `d.GetOk()` would "handle" an empty string:

```suggestion
				ValidateFunc: validation.StringLenBetween(0, 1024),
```